### PR TITLE
feat: add lms_user_email and content_tile to Transactions

### PR DESCRIPTION
**Pull request by johnnagro**
**Description:**

- Add optional `lms_user_email` and `content_tile` fields to Transactions.
- Setup the project with `make test-shell` cribbed from `edx-enterprise`
 
In support of:
- https://2u-internal.atlassian.net/browse/ENT-7867
- https://2u-internal.atlassian.net/browse/ENT-7868

### CODE REVIEW

File: openedx_ledger/models.py [hunk 1]
```diff
@@ -353,6 +363,16 @@ class Meta:
             "The globally unique content identifier.  Joinable with ContentMetadata.content_key in enterprise-catalog."
         )
     )
+    content_title = models.CharField(
+        max_length=255,
```
Comment (iloveagent57) [inline openedx_ledger/models.py:367]: This may be too short for content titles.

File: openedx_ledger/models.py [hunk 2]
```diff
@@ -344,6 +344,16 @@ class Meta:
...
         ),
     )
+    lms_user_email = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        db_index=True,
+        help_text=(
+            "The email of the Open edX LMS user record with which this Transaction is associated."
+            "The email is captured at the time the Transaction is created may not be up to date."
```
Comment (iloveagent57) [inline openedx_ledger/models.py:354]: nit: missing "and" in "Transaction is created may not"?

File: openedx_ledger/models.py [hunk 3]
```diff
@@ -353,6 +363,16 @@ class Meta:
...
         )
     )
+    content_title = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        db_index=True,
+        help_text=(
+            "The title of the content associated with this Transaction."
+            "The title is captured at the time the Transaction is created may not be up to date."
```
Comment (iloveagent57) [inline openedx_ledger/models.py:373]: nit: missing "and"?

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Docker in this repo is only supported for running tests locally
+# as an alternative to virtualenv natively - johnnagro 2022-02-11
```
Comment (iloveagent57) [inline docker-compose.yml:2]: nit: 2023-10-24

**Review by macdiesel — APPROVED** [review]
Thanks for the testing container.

File: openedx_ledger/models.py (hunk 1 above)
Comment (johnnagro) [reply openedx_ledger/models.py:367]: Looks like discovery uses 255 - https://github.com/openedx/course-discovery/blob/master/course_discovery/apps/course_metadata/models.py#L1319

make it bigger here anyway?

**Pull request merged**